File: scala/difference-of-squares/src/main/scala/DifferenceOfSquares.scala
object DifferenceOfSquares {

  def sumOfSquares(n: Int): Int = series(n).map(square).sum

  def squareOfSum(n: Int): Int = square(series(n).sum)

  def differenceOfSquares(n: Int): Int = squareOfSum(n) - sumOfSquares(n)

  private def series(n: Int) = 1 to n

  private def square(n: Int) = n * n
}

File: typescript/difference-of-squares/difference-of-squares.ts
export default class Squares {
    sumOfSquares: number
    squareOfSums: number
    difference: number

    constructor(n: number) {
        let numbers = []
        for (let i = 1; i <= n; i++) {
            numbers.push(i)
        }

        this.sumOfSquares = numbers.reduce((acc, current) => acc + current * current)

        let sum = numbers.reduce((acc, current) => acc + current)
        this.squareOfSums = sum * sum

        this.difference = this.squareOfSums - this.sumOfSquares
    }


}